packages/back-end/src/routers/data-export/data-export.controller.ts

```diff
@@ -0,0 +1,34 @@
+import type { Response } from "express";
+import { AuthRequest } from "../../types/AuthRequest";
+import { ApiErrorResponse } from "../../../types/api";
+import { getOrgFromReq } from "../../services/organizations";
+import { EventAuditUserForResponseLocals } from "../../events/event-types";
+import { getLatestEventsForOrganization } from "../../models/EventModel";
+import { DataExportFileResponse } from "../../../types/data-exports";
+
+/**
+ * GET /data-export/events
+ * Get one data-export resource by ID
+ * @param req
+ * @param res
+ */
+export const getDataExportForEvents = async (
+  req: AuthRequest,
+  res: Response<
+    DataExportFileResponse | ApiErrorResponse,
+    EventAuditUserForResponseLocals
+  >
+) => {
+  req.checkPermissions("viewEvents");
+
+  const { org } = getOrgFromReq(req);
+
+  const events = await getLatestEventsForOrganization(org.id, 0);
```

Comment:
would it make sense to put some sort of default practical limit or window, possibly let users customize?

Reply:
I discussed that with @jdorn and he said to do this to start.

Another option we had discussed, and it's similar to what Github does, is create some kind of async task that we get updates on, e.g. via polling, in the event the file is so big it cannot be downloaded in a single request. Then when the file is ready, we send a download for it. This would require us to have authentication around accessing static files, which we don't currently have architected.

This approach I've implemented is obviously not going to scale but it's what was asked as a first pass.